pkg/resources/kafka/kafka.go

```diff
@@ -838,11 +839,13 @@ func (r *Reconciler) handleRollingUpgrade(log logr.Logger, desiredPod, currentPo
 		}
 		desiredPod.Spec.Tolerations = uniqueTolerations
 	}
-	// Check if the resource actually updated
+	// Check if the resource actually updated or if labels match TaintedBrokersSelector
 	patchResult, err := patch.DefaultPatchMaker.Calculate(currentPod, desiredPod)
 	switch {
 	case err != nil:
 		log.Error(err, "could not match objects", "kind", desiredType)
+	case r.isPodTainted(log, currentPod):
+		log.Info("pod has tainted labels, deleting it", "pod", currentPod)
```

Comment:
Could there be situations where we need to handle here the case of another pod is already being terminated or still being started?

Reply:
I piggybacked on the existing flow which was already checking pod states and change deltas before reconciling a pod.